src/Illuminate/Session/Middleware/StartSession.php

```diff
@@ -124,6 +124,12 @@ protected function handleStatefulRequest(Request $request, $session, Closure $ne
 
         $this->addCookieToResponse($response, $session);
 
+        foreach (app('db')->getConnections() as $connection) {
+            if ($connection->getConfig('sticky_ttl') && $connection->getRecordModificationState()) {
+                $session->databaseRecordsModified($connection->getName());
+            }
+        }
+
```

Comment:
There seems to be a strange scenario here that I'm not sure how it plays out.

I have a connection configured in `database.connections.tenant` which is lazily configured depending on which tenant the current authenticated user belongs to. Now say User A (tenant-1) modifies the database and then User B (tenant-2) makes a new read-only request.

Because the `database.connections.tenant` is hot-swapped between multiple tenants, the session will now treat it as "this connection has been modified and should use the write database" even though the database is not even the same, just the connection name.

Am I interpreting this right? Should the `database.connections.{connection}.database` perhaps be involved here somehow?

Reply:
Yes it's on the connection level not the DB level. Don't use the feature if you hot swap databases like this.